vignettes/more-recipes.Rmd

```diff
@@ -0,0 +1,277 @@
+center the numerical predictors to allow for meaningful interpretation of the 
+intercept. 
+
+```{r}
+r <- r %>% step_center(contains("lag"), role = "predictor")
+```
+
+Since the current outcomes are positive, but the linear regression assumes the 
+outcome to be in full support, we will consider a log-transformation with an 
+offset value of 1 to ensure `log(0)` does not occur. 
+
+```{r}
+r_lin <- r %>% step_log(ahead_7_deaths, role = "outcome", offset = 1)
+```
+
+Other useful functionalities include creating a set of binary dummy variables 
+from a factor variable. As an illustration that will not affect the models, we 
+will show how states in `geo_value` become dummy variables so that each state 
+as a dummy variable has a differential intercept coefficient.
+
+```{r dummy-variable}
+length(unique(x$geo_value))
+
+dummies <- r %>% 
+  step_dummy(geo_value) %>%
+  prep(training = x) 
+
+dummy_data <- bake(dummies, x)
+
+names(dummy_data)
+```
+
+Once the pre-processing steps are done, we can move on to model fitting using 
+the `parsnip` package. 
+
+
+# Model Fitting and Post-processing
+
+Sometimes, the exact values are desired, therefore fitting a linear regression 
+or poisson regression would be a good idea. 
+
+On the other hand, say instead of predicting 7-day ahead death counts, we would 
+like to classify if the future predictions are high or low based on some given threshold. 
+In this case, a logistic regression among other classification models can be used.
+For the purpose of illustration, we define two categories: less than 10K(low) 
+and more than 10K(high). 
+
+We resort to the `recipes` package again to change the numerical outcomes to 
+categorical for the purpose of classification:
+
+```{r, warning = FALSE}
+binner <- function(x) {
+  x <- cut(x, breaks = c(0, 10000, Inf), include.lowest = TRUE)
+  # now return the group number
+  as.numeric(x)
+}
+
+inc <- c("low", "high")
+
+r_categorical <- r %>% 
+  step_num2factor(ahead_7_deaths, 
+                  role = "outcome", 
+                  transform = binner, 
+                  levels = inc)
+
+```
+
+As a sanity check we can see now that the preprocessed data has a categorical 
+outcome in `ahead_7_deaths`. 
+
+```{r, warning FALSE, echo = FALSE}
+glimpse(slice_sample(bake(prep(r_categorical, x),x), n = 6))
+```
+
+
+### Linear Regression
+
+First, we can try fitting a linear regression:
+
+```{r linear-fit-workflow, warning = FALSE}
+wf <- epi_workflow(r_lin, parsnip::linear_reg()) %>%
+    fit(x) 
+
+latest <- get_test_data(recipe = r_lin, x = x)
+```
+
+We take a look at the linear regression fit and the values of the coefficients.
+
+```{r linear-fit, warning = FALSE}
+wf$fit$fit # look at the model fit and coefficients
+
+p <- predict(wf, latest) %>% filter(!is.na(.pred))
+
+p
+```
+
+The predictions are saved in the `.pred` column and represents the log-transformed 
+predicted death counts 7 days from Dec 31. 2021. 
+
+
+### Poisson regression
```

Comment:
For each model, create its own section. Poisson regression first for original death count data which can be downloaded from the website. Then linear reg with population scales and introduce postprocessing and quantile- prediction first, then the quantile, then the population scaling. Don't reverse the order or bad things will happen.